### PR TITLE
Document differences to plain K8s

### PR DESCRIPTION
**Pull request by mhutter**
These are things that often come up when discussing with customers, so I thought it's worth having them documented.

## Checklist

- [x] Try to isolate changes into separate PRs (to build a better changelog).
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `how-to`, `tutorial`, `reference`, `explanation`, `cicd`, `dependency`
      as they show up in the changelog
- [ ] Link this PR to related issues if applicable.

### CODE REVIEW

**Comment by mhutter** [comment]
Actually might be better suited under "Explanation"

File: docs/modules/ROOT/pages/explanation/kubernetes.adoc [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+This has some implications, for example:
+
+* Images that try to bind to ports < 1024 won't work
+* Images that try to `setuid` during their startup process won't work
+
+Most modern software allows you to configure the ports it binds on, and not take assumptions about user or group IDs.
+For example Nginx, in its default configuration, will try to bind to port 80, and then drop to an unprivileged user using `setuid`/`setgid`.
+In order to run Nginx in OpenShift, its configuration must be amended to bind to ports > 1024, and any `user` or `group` directives removed.
+
+Note that if you are using persistent volumes, you don't have to worry about `chown`ing the contents, OpenShift will do this for you automatically when a volume is mounted.
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/explanation/kubernetes.adoc:33]:
```suggestion
NOTE: If you are using persistent volumes, you don't have to worry about `chown`ing the contents, OpenShift will do this for you automatically when a volume is mounted.
```

File: docs/modules/ROOT/pages/explanation/kubernetes.adoc [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+<1> Mount at whatever path your app needs to write to
+<2> `sizeLimit` is optional
+
+
+[#network-policies]
+== Network Namespace Isolation
+
+[.lead]
+While on many vanilla Kubernetes clusters, all pods can communicate with each other over the network, OpenShift enforces the use of https://kubernetes.io/docs/concepts/services-networking/network-policies/[Network Policies].
+By default, all INCOMING traffic must be allowed explicitly, while all OUTGOING traffic is allowed.
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/explanation/kubernetes.adoc:86]:
```suggestion
On OpenShift, this is only allowed for Pods within the same namespace. Allowing pods from other namespaces to connect, must explicitly be allowed using NetworkPolicies.
```

Comment (corvus-ch) [inline docs/modules/ROOT/pages/explanation/kubernetes.adoc:86]:
```suggestion
On most Kubernetes clusters, including OpenShift, all pods can communicate with each other through the clusters internal network. APPUiO Cloud by default, only allows this for pods within the same namespace. https://kubernetes.io/docs/concepts/services-networking/network-policies/[Network Policies] must be used to allow incomming trafic from other namespaces. Incomming trafic from ingress is always allowed.
```

File: docs/modules/ROOT/pages/explanation/kubernetes.adoc [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+          port: 3000  # <6>
+----
+<1> Give your policy a telling name, so you can identify what it's supposed to do later
+<2> The NetworkPolicy must be created in the DESTINATION namespace
+<3> Select which DESTINATION pods are affected by this NetworkPolicy
+<4> Select which SOURCE namespaces you want to allow traffic from. Labels must be used to select namespaces; the `kubernetes.io/metadata.name` label is added automatically to all namespaces
+<5> Select which specific SOURCE pods in the selected namespaces are allowed
+<6> Configure which ports traffic is allowed from
+
+[WARNING]
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/explanation/kubernetes.adoc:156]:
```suggestion
[CAUTION]
```

See https://asciidoctor.org/docs/asciidoc-writers-guide/#admonitions.

**Review by akosma — APPROVED** [review]

**Review by corvus-ch — CHANGES_REQUESTED** [review]
Thanks for the contribution.

I have added a suggestion to clarify that the behavior of the network is an APPUiO and not on OpenShift thing. As a mater of fact, it is actually a APPUiO Managed OpenShift thing.

File: docs/modules/ROOT/pages/explanation/kubernetes.adoc (hunk 3 above)
Comment (mhutter) [reply docs/modules/ROOT/pages/explanation/kubernetes.adoc:156]: The guide you linked says

>Use WARNING to inform the reader of **danger, harm, or consequences** that exist.

There is the danger to expose your applications to untrusted third parties, which could cause harm or other consequences, so WARNING seems appropriate?

Comment (mhutter) [reply docs/modules/ROOT/pages/explanation/kubernetes.adoc:155]: … but the user must also **act carefully** …

Comment (corvus-ch) [reply docs/modules/ROOT/pages/explanation/kubernetes.adoc:155]: You are probably right. I was more looking at from a "proceed with caution to not mess up" perspective.

**Review by corvus-ch — APPROVED** [review]

**Pull request merged**